devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -111,9 +115,16 @@ private void registerTasks(Project project) {
                             .extendsFrom(configurations.findByName(JavaPlugin.TEST_RUNTIME_ONLY_CONFIGURATION_NAME));
 
                     Task testNative = tasks.create(TEST_NATIVE_TASK_NAME, QuarkusTestNative.class);
-                    testNative.dependsOn(buildNative);
+                    testNative.dependsOn(quarkusBuild);
                     testNative.setShouldRunAfter(Collections.singletonList(tasks.findByName(JavaPlugin.TEST_TASK_NAME)));
 
+                    project.getGradle().getTaskGraph().whenReady(taskGraph -> {
```

Comment:
Isn't this already done in `configureBuildNativeTask(...)`?

Reply:
well, yes, you right, I moved out the logger and used task path instead of task reference.